### PR TITLE
Github issue #359

### PR DESCRIPTION
**Pull request by amirhosv**
*Issue #359*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -669,6 +669,15 @@ add_custom_target(check-junit-AesGcmEager
 
     DEPENDS accp-jar tests-jar)
 
+add_custom_target(check-junit-DifferentTempDir
```
Comment (geedo0) [inline CMakeLists.txt:672]: Not convinced that this is a high signal test. There's no assertion anywhere that a different tmpdir was actually used. Just that these two tests succeeded when this property was set. I'm inclined to drop this test target.

File: README.md [hunk 2]
```diff
@@ -344,6 +344,9 @@ Thus, these should all be set on the JVM command line using `-D`.
   encryption/decryption operations would not require allocation and release of `EVP_CIPHER_CTX`
   structure. A common use case would be having long-running threads that each would get its
   own instance of `Cipher` class.
+* `com.amazon.corretto.crypto.provider.tmpdir`
```
Comment (WillChilds-Klein) [inline README.md:347]: what happens if this points to a directory containing the libs we unpack into it? do they get overwritten or do the writes fail?

Comment (amirhosv) [reply README.md:347]: We actually create a new directory here first and extract the files in there. Upon exit, the directory is marked to be deleted.

**Review by WillChilds-Klein — APPROVED** [review]

**Review by alexw91 — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (amirhosv) [reply CMakeLists.txt:672]: Since the PR already has the two approvals, I'll drop this in another PR.

**Pull request merged**